lib/iris/tests/unit/cube_coord_common/test_CFVariableMixin.py

```diff
@@ -0,0 +1,155 @@
+"""
+Unit tests for the :class:`iris._cube_coord_common.CFVariableMixin`.
+"""
+
+from __future__ import (absolute_import, division, print_function)
+from six.moves import (filter, input, map, range, zip)  # noqa
+
+# Import iris.tests first so that some things can be initialised before
+# importing anything else.
+import iris.tests as tests
+
+from iris._cube_coord_common import CFVariableMixin
+
+
+class Test_token(tests.IrisTest):
+    def test_passthru_None(self):
+        result = CFVariableMixin.token(None)
+        self.assertIsNone(result)
+
+    def test_fail_leading_underscore(self):
+        result = CFVariableMixin.token('_nope')
+        self.assertIsNone(result)
+
+    def test_fail_leading_dot(self):
+        result = CFVariableMixin.token('.nope')
+        self.assertIsNone(result)
+
+    def test_fail_leading_plus(self):
+        result = CFVariableMixin.token('+nope')
+        self.assertIsNone(result)
+
+    def test_fail_leading_at(self):
+        result = CFVariableMixin.token('@nope')
+        self.assertIsNone(result)
+
+    def test_fail_space(self):
+        result = CFVariableMixin.token('nope nope')
+        self.assertIsNone(result)
+
+    def test_fail_colon(self):
+        result = CFVariableMixin.token('nope:')
+        self.assertIsNone(result)
+
+    def test_pass_simple(self):
+        token = 'simple'
+        result = CFVariableMixin.token(token)
+        self.assertEqual(result, token)
+
+    def test_pass_leading_digit(self):
+        token = '123simple'
+        result = CFVariableMixin.token(token)
+        self.assertEqual(result, token)
+
+    def test_pass_mixture(self):
+        token = 'S.imple@one+two_3'
+        result = CFVariableMixin.token(token)
+        self.assertEqual(result, token)
+
+
+class Test_name(tests.IrisTest):
+    def setUp(self):
+        # None token CFVariableMixin
+        self.cf_var = CFVariableMixin()
+        self.cf_var.standard_name = None
+        self.cf_var.long_name = None
+        self.cf_var.var_name = None
+        self.cf_var.attributes = {}
+        self.default = 'unknown'
+        # bad token CFVariableMixin
+        self.cf_bad = CFVariableMixin()
+        self.cf_bad.standard_name = None
+        self.cf_bad.long_name = 'nope nope'
+        self.cf_bad.var_name = None
+        self.cf_bad.attributes = {'STASH': 'nope nope'}
+
+    def test_standard_name(self):
+        token = 'air_temperature'
+        self.cf_var.standard_name = token
+        result = self.cf_var.name()
+        self.assertEqual(result, token)
+
+    def test_long_name(self):
+        token = 'long_name'
+        self.cf_var.long_name = token
+        result = self.cf_var.name()
+        self.assertEqual(result, token)
+
+    def test_var_name(self):
+        token = 'var_name'
+        self.cf_var.var_name = token
+        result = self.cf_var.name()
+        self.assertEqual(result, token)
+
+    def test_stash(self):
+        token = 'stash'
+        self.cf_var.var_name = token
```

Comment:
This doesn't test stash, this repeats the var_name test? I'm guessing this was a copy and paste error?

Reply:
Yup, good spot 👀